src/fhirtypes/CodeSystem.ts

```diff
@@ -0,0 +1,106 @@
+import { Meta } from './specialTypes';
+import { Extension } from '../fshtypes';
+import { Narrative, Resource, Identifier, CodeableConcept, Coding } from './dataTypes';
+import { ContactDetail, UsageContext } from './metaDataTypes';
+import { ValueSet } from './ValueSet';
+
+/**
+ * Class representing a FHIR R4 CodeSystem
+ * @see {@link https://www.hl7.org/fhir/codesystem.html}
+ */
+export class CodeSystem {
+  id?: string;
+  meta?: Meta;
+  implicitRules?: string;
+  language?: string;
+  text?: Narrative;
+  contained?: Resource[];
+  extension?: Extension[];
+  modifierExtension?: Extension[];
+  url?: string;
+  identifier?: Identifier[];
+  version?: string;
+  name?: string;
+  title?: string;
+  status = 'active';
+  experimental?: boolean;
+  date?: string;
+  publisher?: string;
+  contact?: ContactDetail[];
+  description?: string;
+  useContext?: UsageContext[];
+  jurisdiction?: CodeableConcept[];
+  purpose?: string;
+  copyright?: string;
+  caseSensitive?: boolean;
+  valueSet?: ValueSet;
+  hierarchyMeaning?: string;
+  compositional?: boolean;
+  versionNeeded?: boolean;
+  content = 'complete';
+  supplements?: CodeSystem;
```

Comment:
This is actually `canoncial(CodeSystem)`, so, like above, we should just count it as a `string`.